src/features/metadata-instance-editor/MetadataInstanceConfirmDialog.js

```diff
@@ -15,20 +15,33 @@ type Props = {
     onConfirm: Function,
 };
 
-const MetadataInstanceConfirmDialog = ({ onCancel, onConfirm, confirmationMessage }: Props) => (
-    <div className="metadata-instance-confirm-cover">
-        <div className="metadata-instance-confim-container">
-            <p className="metadata-instance-confirm-text">{confirmationMessage}</p>
-            <div className="metadata-instance-confirm-buttons">
-                <Button data-resin-target="metadata-confirmcancel" onClick={onCancel} type="button">
-                    <FormattedMessage {...messages.metadataCancel} />
-                </Button>
-                <PrimaryButton data-resin-target="metadata-confirmremove" onClick={onConfirm} type="button">
-                    <FormattedMessage {...messages.customRemove} />
-                </PrimaryButton>
+const MetadataInstanceConfirmDialog = ({ onCancel, onConfirm, confirmationMessage }: Props) => {
+    const confirmationMessageElement = React.useRef(null);
+
+    React.useEffect(() => {
+        if (confirmationMessageElement.current) {
+            confirmationMessageElement.current.focus();
+        }
+    });
+
+    return (
+        <div className="metadata-instance-confirm-cover">
+            <div className="metadata-instance-confim-container">
+                {/* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */}
+                <div tabIndex="0" ref={confirmationMessageElement}>
```

Comment:
This seems like an unideal solution. Would using something like the `role="alert"` work instead to get the screen reader to read the text without explicitly providing focus?

Reply:
Hi. We tired role='alert' but linter still fail. Do you have any other suggestion?

(Linter says non-interactive elements can't have tab-index property.)

Reply:
I was suggesting that instead of the `tabIndex` approach, whether providing the `role="alert"` on the `MetadataInstanceConfirmDialog` would solve the voiceover issue when it mounts?

Reply:
+1 to @ConradJChan, adding aria-role="alert" would create a live region. The autofocus, if any, should probably go to the Cancel button.

Reply:
We need to have ​autofocus on the message of `metadata-instance-confim-container` in order to VoiceOver read this section.
role="alert" does not solve the problem of VoicerOver. TabIndex does.

Reply:
@alexkrolick we need to focus the message before focus the cancel button. Otherwise VoiceOver will read Cancel but no message

Reply:
Maybe try `role="alert"` instead? I put together a codesandbox that seems to work as expected: https://codesandbox.io/s/confirm-dialog-a11y-29ze3

Reply:
Thanks clarify this. It works!